### PR TITLE
Add support for the teleconsultation record deeplink

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.clubhouse.io/simpledotorg/story/884/add-support-for-the-teleconsultation-record-deeplink

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/deeplink/DeepLinkUpdate.kt [hunk 1]
```diff
@@ -34,14 +34,22 @@ class DeepLinkUpdate : Update<DeepLinkModel, DeepLinkEvent, DeepLinkEffect> {
...
+    val effect = if (event.patient != null) {
+      navigateToPatientSummary(model)
     } else {
       ShowPatientDoesNotExist
     }
     return dispatch(effect)
   }
+
+  private fun navigateToPatientSummary(model: DeepLinkModel): DeepLinkEffect {
+    return if (model.isLogTeleconsultDeepLink) {
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/deeplink/DeepLinkUpdate.kt:49]: Why would we want to remove `isLogTeleconsultDeepLink` later? We still would want to check if navigating to `NavigateToPatientSummaryWithTeleconsultLog` or `NavigateToPatientSummary`, right ?

Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/deeplink/DeepLinkUpdate.kt:49]: WDYM? We will continue using it 🤔

Comment (Honey14) [reply app/src/main/java/org/simple/clinic/deeplink/DeepLinkUpdate.kt:49]: Oops, my mistake, there would PatientSummaryScreen with Teleconsult log only.

**Review by Honey14 — APPROVED** [review]
Looks good

**Pull request merged**